app/scripts/services/Room.js

```diff
@@ -1,15 +1,42 @@
 (function() {
   function Room($firebaseArray) {
-    var ref = firebase.database().ref().child("rooms");
-    var rooms = $firebaseArray(ref);
+
+    var roomsRef = firebase.database().ref().child("rooms");
+    var messagesRef = firebase.database().ref().child("messages");
+    var rooms = $firebaseArray(roomsRef);
+    var messages = $firebaseArray(messagesRef);
+
     var roomService = {};
 
 
     roomService.all = rooms;
+
     roomService.addRoom = function (roomTitle) {
-      rooms.$add({name: roomTitle});
+      rooms.$add({
+        name: roomTitle,
+        created_at: firebase.database.ServerValue.TIMESTAMP
+      });
+    };
+
+    roomService.addMessage = function(userName, content, roomId) {
+
+      messages.$add({
+        username: userName,
+        content: content,
+        sent_at: firebase.database.ServerValue.TIMESTAMP,
+        roomid: roomId
+      });
     };
-    
+
+    roomService.getMessages = function(room) {
+
+       var query = messagesRef.orderByChild('roomid').equalTo(room.$id);
+       var queryArray = $firebaseArray(query);
+
+       return queryArray;
```

Comment:
This is totally cool the way you have it (for readability) but FYI both `query` and `queryArray` are both only used once, so you could rewrite this into a single line if so desired.